### ISSUE TITLE
Headers not set correctly

### ISSUE THREAD

**Issue by KuenzelIT** [FriendsOfPHP/Goutte#260]
Hi,

when I use 

```php
$this->client->setHeader("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:45.0) Gecko/20100101 Firefox/45.0");
```

I get two User-Agents in my header when looking at it with WireShark:
```
user-agent: Symfony2 BrowserKit\r\n
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:45.0) Gecko/20100101 Firefox/45.0\r\n
```

Looks like somewhere deep is strtolower used, but not for the custom headers.

Not sure if this is Gouttes "fault" or a fault in Guzzle/BrowserKit/...

I'm not too familiar with Goutte/Guzzle/..., so what do you guys think?

**Comment by 3zzy** [comment]
This does the trick: 

```
$client->setHeader('user-agent', $ua);
```

.. notice that `user-agent` is all lowercase.

**Comment by KuenzelIT** [comment]
Yeah, that's right, but it shouldn't work that way right? It should compare case insensitive and replace the default header.
